Content.Server/Damage/Events/StaminaMeleeHitEvent.cs

```diff
@@ -0,0 +1,30 @@
+using Robust.Shared.Collections;
+using Content.Server.Damage.Components;
+
+namespace Content.Server.Damage.Events;
+
+/// <summary>
+/// The components in the list are going to be hit,
+/// give opportunities to change the damage or other stuff.
+/// </summary>
+public sealed class StaminaMeleeHitEvent : HandledEntityEventArgs
```

Comment:
why not use/modify `MeleeHitEvent` and check if the base damage is stamina instead of this?

Reply:
stamina damage isn't damage